src/lib/hierarchy-operations/index.js

```diff
@@ -167,9 +194,26 @@ function replaceLineageInSingleContact(doc, moveContext) {
   }
 }
 
-function replaceLineageInContacts(moveContext) {
+function updateContacts(moveContext, constraints) {
   return moveContext.descendantsAndSelf
-    .map(descendant => replaceLineageInSingleContact(descendant, moveContext))
+    .map(descendant => {
+      const deleteSource = moveContext.merge && descendant._id === moveContext.sourceId;
+      const deletePrimaryContact = moveContext.mergePrimaryContacts 
+        && descendant._id === moveContext.sourcePrimaryContactId 
+        && moveContext.destinationPrimaryContactId;
+
+      if (deleteSource || deletePrimaryContact) {
+        const toDeleteUsers = moveContext.disableUsers && constraints.isPlace(descendant);
+        return {
+          _id: descendant._id,
+          _rev: descendant._rev,
```

Comment:
issue(non-blocking): Similar to [this comment](https://github.com/medic/cht-conf/pull/649#discussion_r1881581392), can you check if `_rev` is required?

Reply:
yes, to delete a doc both _id and _rev are required together with _deleted: true